addons/stock/models/product_strategy.py

```diff
@@ -27,6 +27,10 @@ def _default_category_id(self):
     def _default_location_id(self):
         if self.env.context.get('active_model') == 'stock.location':
             return self.env.context.get('active_id')
+        if self.env['stock.warehouse'].search_count([]) == 1:
```

Comment:
You can check the group multi-warehouse (`stock.group_stock_multi_warehouses`) instead (If you activate two company having a single warehouse both, your check will failed).